Post AHM Cleanup: Remove Staking (and friends) from Westend Runtime.

Ank4n opened the pull request:
- Remove following pallets from Westend runtime: pallet_election_provider_multi_phase, pallet_bags_list, pallet_nomination_pools, pallet_fast_unstake, pallet_delegated_staking (including weights and other related code such as bag generation logic)
- Ensure try runtime checks pass for Westend (without any [excluded pallets](https://github.com/paritytech/polkadot-sdk/commit/463ef05cf1b5dbbcd91379a952d0ba0f0af347bb#diff-5264d812f7aadfc486efffd3322b11c8dee5e8ddf2e77e0a9e70b573f3d2afddR2829)). 
- Retain the `ProxyType::Staking` to allow setting session key via it.

Earlier CI failure https://github.com/paritytech/polkadot-sdk/actions/runs/19711056203/job/56471786443#step:9:534 is also fixed by retaining now-deprecated enum variants which prevented correct serialization / deserialization. We initially suspected that [this was not properly fixed](https://github.com/paritytech/polkadot-sdk/issues/8717) and Westend still had some staking holds, but this turned out to not be related.

Closes https://github.com/paritytech/polkadot-sdk/issues/9442

Ank4n committed:
remove try state checks from legacy staking pallet

Ank4n committed:
missing ok

Ank4n committed:
remove unused pallets from westend

Ank4n committed:
fmt

Ank4n committed:
move westend inflation tests to ah

Ank4n committed:
fmt

Ank4n committed:
fix compile

Ank4n committed:
re-enable check runtime migration

Ank4n committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

Ank4n committed:
fix polkadot-voter-bags cli

kianenigma reviewed (dismissed):
Only major point is retaining staking proxy. Even though the name doesn't make perfect sense anymore, it is useful.

Ank4n committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

github-actions[bot] committed:
Update from github-actions[bot] running command 'prdoc --force'

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
Update tests to use new issuance curve

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
More clean-up

andreitrand commented:
/cmd prdoc --audience runtime_dev

andreitrand committed:
CI fixes #1

andreitrand committed:
CI fixes #2

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
CI fixes #3

We remove benchmarking logic that was strongly tied to the pre-migration
staking dependency (impl pallet_staking::Config for Runtime) which has
just been removed.

andreitrand committed:
CI fixes #4

Remove a test that doesn't compile anymore because pallet_staking
is no longer part of Westend.

sigurpol commented:
> Only major point is retaining staking proxy. Even though the name doesn't make perfect sense anymore, it is useful.

Maybe not anymore, now we can set session keys on AH so  on  Westend we could be more aggressive and deprecate setting keys via RC already and so get rid of proxy staking too.  Might be a followup PR  too

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
CI fixes #5

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
CI fixes #6

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
Fix umbrella

andreitrand committed:
CI fixes #7

andreitrand commented on polkadot/runtime/westend/src/bag_thresholds.rs:
```diff
@@ -28,207 +28,3 @@ pub const EXISTENTIAL_WEIGHT: u64 = 10_000_000_000;
 #[cfg(any(test, feature = "std"))]
 #[allow(unused)]
 pub const CONSTANT_RATIO: f64 = 1.1131723507077667;
```
Yes, it has been removed.

andreitrand commented on polkadot/runtime/westend/src/lib.rs:
```diff
@@ -1311,25 +1070,10 @@ impl InstanceFilter<RuntimeCall> for ProxyType {
-				RuntimeCall::FastUnstake(..)
+				RuntimeCall::Auctions(..) // Specifically omitting the entire XCM Pallet
 			),
-			ProxyType::Staking => {
```
Yes, I kept it for now. We can remove it in a later PR.

andreitrand commented:
> > Only major point is retaining staking proxy. Even though the name doesn't make perfect sense anymore, it is useful.
> 
> Maybe not anymore, now we can set session keys on AH so on Westend we could be more aggressive and deprecate setting keys via RC already and so get rid of proxy staking too. Might be a followup PR too

Let's keep it it for now and remove it in a subsequent, dedicated PR in order to reduce the potential blast radius of this PR.

andreitrand committed:
CI fixes #8

andreitrand commented on polkadot/runtime/westend/src/lib.rs:
IIRC, `MaxAuthorities` is still needed as-is for block authoring and consensus. I've set `type MaxNominators` to `ConstU32<0>` and removed the corresponding constant since they no longer appear to be needed by the remaining RC staking logic.

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
Update prdoc

andreitrand committed:
Update prdoc #2

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand committed:
Merge branch 'master' into ankn-staking-remove-trystate-checks

andreitrand commented:
Closing this PR as it has been recreated in https://github.com/paritytech/polkadot-sdk/pull/11940.
